File: bots/javaClient/src/main/java/cb/bots/FirstAI.java
package cb.bots;

import cb.PlayState;

public class FirstAI implements BotAi {
    private String name = "JavaTestBot";
    private boolean fly = true;

    public FirstAI() {
    }

    @Override
    public boolean play(PlayState playState) {
        if (playState.getPlayer().getPosY() > 450) {
            fly = true;
        }

        if (playState.getPlayer().getPosY() < 50) {
            fly = false;
        }

        return fly;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: bots/typescriptClient/src/bots/firstAI.ts
import { BotAI } from './botAI';
import { PlayState } from './data';

export class FirstAI implements BotAI {
  private name: string = 'TypeScriptTestBot';
  private fly: boolean = true;

  play(currentGameState: PlayState): boolean {
    console.log(currentGameState);
    if (currentGameState.player.pos_y > 450) {
      this.fly = true;
    }

    if (currentGameState.player.pos_y < 50) {
      this.fly = false;
    }

    return this.fly;
  }

  getName(): string {
    return this.name;
  }
} 